sw/device/tests/crypto/ecdsa_p256_functest.c

```diff
@@ -34,23 +34,15 @@ static const ecdsa_p256_private_key_t kPrivateKey = {
     .d1 = {0},
 };
 
-hmac_error_t compute_digest(void) {
+static void compute_digest(void) {
   // Compute the SHA-256 digest using the HMAC device.
   hmac_sha256_init();
-  hmac_error_t err = hmac_sha256_update(&kMessage, sizeof(kMessage) - 1);
-  if (err != kHmacOk) {
-    return err;
-  }
+  CHECK_STATUS_OK(hmac_sha256_update(&kMessage, sizeof(kMessage) - 1));
```

Comment:
nit (not PR blocking): Does this result in an escalation? It would be good to define how we want to respond to errors when the library is called from TockOS land.

Reply:
This particular one (`CHECK_STATUS_OK`) prints a `CHECK-fail` message and ends the test with a "failure" status. It won't get called from TockOS I think, since it's test-only.

For non-test code, `HARDENED_TRY` will propagate the error up all the way back to the caller. There are two types of "operation failed" errors: recoverable and fatal. For recoverable errors, it should be okay for TockOS to just retry the same operation (although there is no guarantee a retry will succeed). After a fatal error, TockOS should not retry and should assume one or more hardware blocks are locked and we need a reset, and that we may be under attack.